### PR TITLE
Bleve search doc updates

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation ticket: https://mattermost.atlassian.net/browse/MM-37211

Updated:
- Install, Deploy, Upgrade, and Scale > Upgrade Mattermost > Upgrading Mattermost Server
   - Updated the Bleve notes to clarify that Bleve indexes can be migrated without reindexing between MM versions
   - Updated page formatting to improve readability and clarity of process steps

### CODE REVIEW

File: source/upgrade/upgrading-mattermost-server.rst [hunk 1]
```diff
@@ -21,151 +21,138 @@ Read these instructions carefully from start to finish. Make sure that you under
...
+  By default, your data directories will be preserved with the following commands, including ``config``, ``logs``, ``plugins``, ``client/plugins``, and ``data`` (unless you have a different value configured for local storage).
+
+Custom directories are any directories that you've added to Mattermost and are not preserved by default. Generally, these are TLS keys or other custom information.
+
+If using `Bleve Search <https://docs.mattermost.com/deploy/bleve-search.html>`__, the index directory path won't be preserved with the commands below if the directory exists *within* the ``mattermost`` directory. 
+  
+  - You can either move the bleve index directory out from the ``mattermost`` directory before upgrading or, following an upgrade, you can copy the contents of the bleve index directory from the ``backup`` directory. 
+  - You can then store that directory or re-index as preferred. 
+  - The Bleve indexes can be migrated without reindexing between Mattermost versions. See our `Configuration Settings <https://docs.mattermost.com/configure/configuration-settings.html#bleve-settings-experimental>`__ documentation for details on setting the bleve index directory.
```
Comment (justinegeffen) [inline source/upgrade/upgrading-mattermost-server.rst:102]:
```suggestion
  - The bleve indexes can be migrated without reindexing between Mattermost versions. See our `Configuration Settings <https://docs.mattermost.com/configure/configuration-settings.html#bleve-settings-experimental>`__ documentation for details on setting the bleve index directory.
```

**Review by justinegeffen — APPROVED** [review]
Some minor suggestions, non-blocking other than one typo. Approving for expediency.

File: source/upgrade/upgrading-mattermost-server.rst [hunk 2]
```diff
@@ -21,151 +21,138 @@ Read these instructions carefully from start to finish. Make sure that you under
...
 
-            $ ls /opt/mattermost
-            ENTERPRISE-EDITION-LICENSE.txt README.md  client  data   i18n  manifest.txt  prepackaged_plugins
-            NOTICE.txt                      bin        config  fonts  logs  plugins       templates
-          
-    b. Identify if any custom directories from the above step need to be preserved. For each custom directory within the Mattermost folder that you wish to preserve add ``-o -path  mattermost/yourFolderHere`` to the below commands. See the example below where the folder ``yourFolderHere`` is preserved by adding ``-o -path  mattermost/yourFolderHere``.
-    
       .. code-block:: sh
+         sudo find mattermost/ mattermost/client/ -mindepth 1 -maxdepth 1 \! \( -type d \( -path mattermost/client -o -path mattermost/client/plugins -o -path mattermost/config -o -path mattermost/logs -o -path mattermost/plugins -o -path mattermost/data -o -path  mattermost/yourFolderHere \) -prune \) | sort | xargs echo rm -r
```
Comment (PeterDaveHello) [inline source/upgrade/upgrading-mattermost-server.rst:107]: Some content like this line is missing, not rendered after this change ... the docs is broken now 😱

**Comment by PeterDaveHello** [comment]
Please take a look at the rich diff of the rendered result:
https://github.com/mattermost/docs/pull/4924/files?short_path=fcc1287#diff-fcc12871eb71ce483f8dbcabf9b5a0a79ae12f56e7b91a91b6fd3fba83b1a3bd

A lot of useful content was deleted, and I believe it's by accident.

**Comment by cwarnermm** [comment]
Thank you for letting us know about this issue! This content was removed by mistake and is being added back via https://github.com/mattermost/docs/pull/4936.

**Pull request merged**